NaryzhnyDI/lab_3(part_1)/main.cpp

```diff
@@ -0,0 +1,180 @@
+        print();
+    }
+    void u_error(int &x) const{
+        if(x>=size){
+            cout <<"Ошибка: по индексу " << x <<" нельзя добавить или удалить элемент, т.к. индекс последнего "
+                                                "элемента в массиве: " << size-1 << endl;
+        }
+        if(x<0){
+            cout <<"Ошибка: по индексу " << x << " нельзя добавить или удалить элемент"
+                                                 " т.к. нумерация в массиве с 0" << endl;
+        }
+    }
+public:
+    Vector(int u_size){
+        if(u_size>=0) {
+            size = u_size;
+            capacity = size * 2;
+            arr = new int[capacity];
+            fullfill();
+            print();
+        }
+        else{
+            cout <<"Ошибка: массив не может состоять из <0 элементов." << endl;
+        }
+    }
+    void u_push_back() {
+        int temp_i(0);
+        cout << "Напишите элемент, который вы хотите добавить в конец массива: " << endl;
+        cin>>temp_i;
+        if (size < capacity) {
+            arr[size] = temp_i;
+            size++;
+        }
+        else{
+            increase_capacity();
+            arr[size] = temp_i;
+            size++;
+        }
+        print();
+    }
+    void u_insert(){
+        if (size<capacity) {
+            int u_val(0);
+            int u_ind(0);
+            cout << "Введите элемент, а затем его желаемую позицию: " << endl;
+            cin >> u_val;
+            cout <<"Теперь позицию (индекс): " << endl;
+            cin >> u_ind;
+            if (size < capacity) {
+                if (u_ind >= 0 && u_ind <= (size-1)) {
+                    insert_element(u_val, u_ind);
+                }
+                else {
+                    u_error(u_ind);
+                }
+            }
+            else {
+                increase_capacity();
+                insert_element(u_val,u_ind);
+            }
+        }
+    }
+    void u_delete(){
+        int u_ind(0);
+        cout <<"Введите индекс элемента, который хотите удалить: " << endl;
+        cin >> u_ind;
+        if (u_ind>=0 && u_ind <= (size-1)){
+            int *temp_arr = new int[capacity];
+            for (int i =0; i< u_ind;i++){
+                temp_arr[i] = arr[i];
+            }
+            arr[u_ind] = 0;
+            for (int j = (u_ind+1);j<size;j++){
+                temp_arr[j-1]=arr[j];
+            }
+            for (int z = 0; z<capacity;z++){
+                arr[z] = temp_arr[z];
+            }
+            delete [] temp_arr;
+            (size--);
+        }
+        else{
+            u_error(u_ind);
+        }
+        print();
+    }
+    void print(){
+        cout<<"Ваш массив сейчас выглядит так:\n";
+        if (size!=0) {
+            for (int j = 0; j < size; j++) {
+                cout << arr[j] << " ";
+            }
+            cout << "\n________ \n";
+        }
+        else{
+            cout << "Ваш массив 0-го размера \n";
+        }
+    }
+    void GetReference(){
```

Comment:
"Getting element by reference using index" means that function should return reference to some element. It seems like just seeing element's address in console output won't satisfy users of your class